doc/sfp-cmis/CMIS_custom_App_Sel.md

```diff
@@ -0,0 +1,212 @@
+
+## 1 Introduction and Scope
+Different electrical link lengths require different module settings.  
+Short mode: Lower output voltage, used for C2M (Chip-to-Module) short-reach links.  
+Long mode: Higher output voltage, used for C2M long-reach links.  
+This impacts performance and helps avoid signal degradation for long-reach connections.
+
+### 1.1 IEEE 802p3ck Requirements
+IEEE Standard for Ethernet: Physical Layer Specifications and Management Parameters for 100 Gb/s, 200 Gb/s, and 400 Gb/s Electrical Interfaces Based on PAM4 signaling for 100 Gb/s.
+It also defines 100Gb/s per lane electrical interface and specifies TP4 output characteristics for signal integrity.
+
+Reference from IEEE 
+
+![image](https://github.com/user-attachments/assets/c082e321-d020-4c1c-992b-00b590759e78)
+
+
+### 1.2 TP4
+TP4 (Test Point 4) is a reference point in high-speed electrical signaling, particularly in IEEE 802.3 Ethernet standards. 
+It is used to define signal integrity requirements at the interface between an electrical host and an optical module (C2M - Chip-to-Module).
+
+TP4 is the measurement point at the output of the module, where the signal is sent towards the host receiver. It is a key location for verifying signal integrity and compliance with IEEE 802.3ck specifications.
+
+ Test Points in the IEEE 802.3 Standard:
+1. TP0 – Transmitter input (inside the chip).
+2. TP1a/TP1 – Input to the module (near-end of the module receiver).
+3. TP2 – Inside the module (before the optical conversion).
+4. TP3 – Optical output of the module (before optical fiber transmission).
+5. TP4 – Electrical output of the module (signal going to the host receiver).
+
+```
++----------------+                     +----------------------+
+|   Host ASIC    |                     |   Optical Module     |
+| (Electrical    |                     | (Electrical + Optical|
+|    Side)       |                     |       Side)          |
+|                |                     |                      |
+|   TX  -------> |  ------TP0--------> |  RX                  |
+|   RX  <------- |  <-----TP4--------- |  TX                  |
++----------------+                     +----------------------+
+```
+
+## 2 Requirements
+This feature would be enabled on a per-platform basis. If a platform wants to use this feature, it must provide the optics_si_app_sel.json file during initialization for XCVRD to parse. The mode value will depend on the platform's TP4 measurement points. The CMIS state machine will be modified to support host-defined APP SEL and program the module EEPROM accordingly.
+
+Modules that do not support CMIS and are not part of the CMIS state machine are out of the scope of this document.
+
+NOTE:  
+This feature modifies the TP4 settings, which can also be achieved by providing the optics_si_settings.json file. Consequently, it will modify the TX/RX Signal Integrity Controls on page 10h with the Explicit Control bit set.   
```

Comment:
@AnoopKamath It seems to me that you are using this feature as an alternative approach to change the TP4 SI settings?